src/matlab/create_nlp_nosnoc.m

```diff
@@ -214,786 +83,69 @@
-comp_res_fesd = Function('comp_res_fesd',{w},{J_comp_fesd});
-comp_res_std = Function('comp_res_std',{w},{J_comp_std});
+w = problem.w; % vectorize all variables, TODO: again, further cleanup necessary
+g = problem.g; % vectorize all constraint functions
+p = problem.p;
+J_fun = problem.cost_fun;
+comp_res = problem.comp_res;
+comp_res_fesd = problem.comp_fesd;
+comp_res_std = problem.comp_std;
 
 %% NLP Solver
-prob = struct('f', J, 'x', w, 'g', g,'p',p);
-solver = nlpsol(solver_name, 'ipopt', prob,opts_ipopt);
+prob = struct('f', problem.cost, 'x', w, 'g', g,'p',p);
+solver = nlpsol(settings.solver_name, 'ipopt', prob, settings.opts_ipopt);
 
 %% Define CasADi function for the switch indicator function.
-nu_fun = Function('nu_fun', {w,p},{nu_vector});
-
-%% settings update
-settings.right_boundary_point_explicit  = right_boundary_point_explicit;
+nu_fun = Function('nu_fun', {w,p},{problem.nu_vector});
 
 %% Outputs
 model.prob = prob;
+model.problem = problem;
 model.solver = solver;
 model.g =  g;
 model.w =  w;
 model.p =  p;
-model.J = J;
+model.J = problem.cost;
 model.J_fun = J_fun;
 model.comp_res = comp_res;
 model.comp_res_fesd = comp_res_fesd;
 model.comp_res_std = comp_res_std;
 model.nu_fun = nu_fun;
 
 % create CasADi function for objective gradient.
-nabla_J = J.jacobian(model.w);
+nabla_J = problem.cost.jacobian(model.w);
 nabla_J_fun = Function('nabla_J_fun', {w,p},{nabla_J});
 model.nabla_J = nabla_J;
 model.nabla_J_fun = nabla_J_fun;
 
 % TODO: make member function
-if print_level > 5
-    disp("g")
-    print_casadi_vector(g)
-    disp('lbg, ubg')
-    disp([length(problem.lbg), length(problem.ubg)])
-    disp([problem.lbg, problem.ubg])
-
-    disp("w")
-    print_casadi_vector(w)
-    disp('lbw, ubw')
-    disp([problem.lbw, problem.ubw])
-    disp('w0')
-    disp(problem.w0)
-
-    disp('objective')
-    disp(J)
+if settings.print_level > 5
+    problem.print();
 end
 
 %% Model update: all index sets and dimensions
 % TODO: Maybe just return the problem, currently trying not to break compatibility for now.
-model.ind_x = problem.ind_x;
-model.ind_elastic = problem.ind_elastic;
-model.ind_v = problem.ind_v;
-model.ind_z = problem.ind_z;
+model.ind_x = [problem.ind_x0.'; flatten_ind(problem.ind_x)];
+model.ind_v = sort(flatten_ind(problem.ind_v));
+model.ind_z = problem.ind_z; %TODO fix this by breaking compat
 model.ind_u = problem.ind_u;
-model.ind_h = problem.ind_h;
-model.ind_g_clock_state = problem.ind_g_clock_state;
-model.ind_sot = problem.ind_sot;
+model.ind_h = flatten_ind(problem.ind_h);
+model.ind_sot = flatten_ind(problem.ind_sot);
 model.ind_t_final  = problem.ind_t_final;
-model.ind_boundary = problem.ind_boundary;
-model.n_cross_comp = n_cross_comp;
-model.h = h;
-model.h_k = h_k;
-model.p_val = p_val;
-model.n_cross_comp_total = sum(n_cross_comp(:));
+model.p_val = problem.p0;
+
+model.structured_ind.x = problem.ind_x;
+model.structured_ind.x0 = problem.ind_x0;
```

Comment:
why are the indices copied to model?
IMO `model` should be agnostic of `problem` stuff.
`problem` and these indices are very specific to discretization etc.

Reply:
Yep this probably should not live in model.